parts/kubernetesagentcustomdata.yml

```diff
@@ -128,7 +128,8 @@ write_files:
   owner: "root"
   content: |
     #!/bin/bash
-    exit 0
+    echo $(curl -q http://169.254.169.254/metadata/v1/InstanceInfo 2>/dev/null | sed "s/.*\"FD\":\"\([0-9]*\).*/\1/") > /etc/kubernetes/fd
```

Comment:
1. This is the metadata server IP
2. This is a well-known non routable IP address that can be accessed only from within the Azure VM. 
3. Thanks, will add "--retry" flag to curl